internal/explorer/db/postgres.go

```diff
@@ -0,0 +1,593 @@
+package db
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	// to use for database/sql
+	_ "github.com/lib/pq"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	// ErrNodeNotFound node not found
+	ErrNodeNotFound = errors.New("node not found")
+	// ErrFarmNotFound farm not found
+	ErrFarmNotFound = errors.New("farm not found")
+)
+
+const (
+	setupPostgresql = `
+	CREATE TABLE IF NOT EXISTS node_pulled (
```

Comment:
You need to also create index on all the values used in the filtering otherwise doing a SELECT on this table is not going to be performant. for example, used_mru and such.

I would also actually keep track of free capacity (not used) so when u do a select on (for example), node that has enough free memory we don't have to do total-used because then this will cause a table scan.